core/state/state_object.go

```diff
@@ -171,9 +219,8 @@ func (c *StateObject) ReturnGas(gas, price *big.Int) {}
 
 func (self *StateObject) Copy(db trie.Database) *StateObject {
 	stateObject := NewStateObject(self.address, db)
-	stateObject.balance.Set(self.balance)
-	stateObject.codeHash = common.CopyBytes(self.codeHash)
-	stateObject.nonce = self.nonce
+	stateObject.data = self.data
+	stateObject.data.Balance.Set(self.data.Balance)
```

Comment:
But `stateObject.data` is the same pointer as `seld.data` so you're updating the same big.Int, no?  You never made a new big.Int